safe-transaction-client/src/lib.rs

```diff
@@ -0,0 +1,343 @@
+    pub value: U256,
+    pub data: Bytes,
+    pub operation: Operation,
+    pub nonce: U256,
+}
+
+#[derive(Debug)]
+pub struct SignedSafeTx {
+    inner: SafeTx,
+    sender: Address,
+    safe_tx_hash: TxHash,
+    signature: Signature,
+}
+
+impl Client<'_> {
+    pub fn new(transactions_api: &str) -> Client {
+        Client {
+            agent: ureq::Agent::new(),
+            transactions_api,
+        }
+    }
+
+    pub fn with_safe(&self, safe_address: Address) -> Safe {
+        Safe {
+            client: self,
+            safe_address,
+        }
+    }
+}
+
+impl Safe<'_> {
+    pub fn create_transaction(
+        &self,
+        to: Address,
+        value: U256,
+        data: Bytes,
+        operation: Operation,
+    ) -> Result<SafeTx> {
+        #[derive(serde::Deserialize)]
+        struct SafeResponse {
+            nonce: u64,
+        }
+
+        let SafeResponse { nonce, .. } = self
+            .client
+            .agent
+            .get(&format!(
+                "{}/v1/safes/{}/",
+                self.client.transactions_api,
+                to_checksum(&self.safe_address, None),
+            ))
+            .call()?
+            .into_json()?;
+
+        Ok(SafeTx {
+            safe_address: self.safe_address,
+            to,
+            value,
+            data,
+            nonce: U256::from(nonce),
+            operation,
+        })
+    }
+
+    pub fn propose(&self, signed_safe_tx: SignedSafeTx) -> Result {
+        let SafeTx {
+            to,
+            value,
+            data,
+            nonce,
+            operation,
+            ..
+        } = signed_safe_tx.inner;
+        self.client
+            .agent
+            .post(&format!(
+                "{}/v1/safes/{}/multisig-transactions/",
+                self.client.transactions_api,
+                to_checksum(&self.safe_address, None),
+            ))
+            .send_json(ureq::json!({
+                "to": to_checksum(&to, None),
+                "value": format!("{}", value),
+                "data": data,
+                "operation": operation as u8,
+                "gasToken": None::<()>,
+                "safeTxGas": "0",
+                "baseGas": "0",
+                "gasPrice": "0",
+                "refundReceiver": Address::zero(),
+                "nonce": format!("{}", nonce),
+                "contractTransactionHash": signed_safe_tx.safe_tx_hash,
+                "sender": to_checksum(&signed_safe_tx.sender, None),
+                "signature": format!("0x{}", signed_safe_tx.signature),
+                "origin": "An unauthenticated origin string?",
+            }))
+            // 200 OK, is not okay. We expect 201 Created.
+            .and_then(|r| match r.status() {
+                201 => Ok(()),
+                s => Err(ureq::Error::Status(s, r)),
+            })
+            .map_err(From::from)
+    }
+
+    pub fn add_to_confirmations(&self, signed_safe_tx: SignedSafeTxHash) -> Result {
```

Comment:
If we're keeping with `propose`, maybe we call this `confirm`? Alternatively I'd rename both to `propose_transaction` and `confirm_transaction` if we want to be explicit.

Reply:
The whole crate is about transactions. That'd kinda be like a `Vec::push_element` instead of `Vec::push`. I'll rename it to `confirm`.